File: leetcode/3351-maximize-happiness-of-selected-children/solution.rs
impl Solution {
    pub fn maximum_happiness_sum(mut happiness: Vec<i32>, k: i32) -> i64 {
        happiness.sort();
        happiness.into_iter()
            .rev()
            .enumerate()
            .map(|(i,x)| (x - i as i32) as i64)
            .take(k as usize)
            .take_while(|&x| x > 0)
            .sum()
    }
}

File: leetcode/3351-maximize-happiness-of-selected-children/solution.go
func maximumHappinessSum(happiness []int, k int) int64 {
	sort.Sort(sort.Reverse(sort.IntSlice(happiness)))
    ret := int64(0)
    for i:= 0;i<k;i++ {
        ret += int64(max(0, happiness[i] - i))
    }
    return ret
}
